File: 214/projects/04/java/Factorial.java
/* Factorial.java outputs n! of the input number
 *
 * Input: double n
 * Output: The factorial of the number
 * Duncan Van Keulen on 2-27-2020
 ********************************************************/
import java.util.Scanner;

class Factorial {
    public static void main(String[] args) {
        Scanner keyboard = new Scanner(System.in);
        System.out.print("Enter a number: ");
        double n; double answer = 1.0;

        n = keyboard.nextDouble();

        // Loop from 2 to n, multiplying each time to calculate factorial
        for(int i = 2; i <= n; i++) {
            answer *= i;
        } 
        System.out.println(n + "! = " + answer);
    }
}

File: 214/projects/04/clojure/src/factorial.clj
;;;; factorial displays the factorial of a number
;;;;
;;;; Input: n, a number
;;;; Precondition: n is a positive number
;;;; Output: The factorial of the input number
;;;;
;;;; By: Duncan Van Keulen
;;;; Date: 2-27-2020
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns factorial)

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; factorial computes n!
;;; Input: n, a positive number
;;; Output: The factorial of that number to the console
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn factorial [n] 
    (def ^:dynamic answer 1)
    (loop [count 2]
      when ((<= count n)
        (* answer count)
        (println answer count)
        (recur (+ count 1))))
    (print n) (print "! = ")
    (println answer)
)
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; -main is a 'driver' for the factorial function.
;;; Input: n, a positive number
;;; Output: The factorial of that number
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;


(defn -main [] 
    (print "Enter a number: ") (flush)
    (def number)
    (let 
        [ number (double (read))]
    )
    (factorial number)

)